modular_sand/code/datums/interactions/interaction_datums/lewd/frotting.dm

```diff
@@ -41,5 +44,7 @@
 						'modular_sand/sound/interactions/squelch2.ogg',
 						'modular_sand/sound/interactions/squelch3.ogg'), 70, 1, -1)
 	user.visible_message("<span class='lewd'><b>\The [user]</b> [message]</span>", ignored_mobs = user.get_unconsenting())
-	partner.handle_post_sex(NORMAL_LUST, CUM_TARGET_VAGINA, user)
-	user.handle_post_sex(NORMAL_LUST, CUM_TARGET_VAGINA, partner)
+	if(partner.can_penetrating_genital_cum())
+		partner.handle_post_sex(NORMAL_LUST, CUM_TARGET_VAGINA, user)
+	if(user.can_penetrating_genital_cum())
+		user.handle_post_sex(NORMAL_LUST, CUM_TARGET_VAGINA, partner)
```

Comment:
Why the check here? This interaction doesn't need penises, it's related to vaginas